pkg/api/helpers.go

```diff
@@ -41,3 +44,31 @@ func GetFieldLabelConversionFunc(supportedLabels map[string]string, overrideLabe
 func GetResourceKey(obj metav1.ObjectMeta) string {
 	return fmt.Sprintf("%s/%s", obj.Namespace, obj.Name)
 }
+
+// OriginObjects returns the origin types.
+func OriginObjects(objs []runtime.Object) []runtime.Object {
+	ret := []runtime.Object{}
+	for _, obj := range objs {
+		if !UpstreamType(obj) {
+			ret = append(ret, obj)
+		}
+	}
+	return ret
+}
+
+// UpstreamObjects returns the non-origin types.
+func UpstreamObjects(objs []runtime.Object) []runtime.Object {
+	ret := []runtime.Object{}
+	for _, obj := range objs {
+		if UpstreamType(obj) {
+			ret = append(ret, obj)
+		}
+	}
+	return ret
+}
+
+// UpstreamType returns true for Kubernetes types.
+func UpstreamType(obj runtime.Object) bool {
+	t := reflect.TypeOf(obj).Elem()
+	return strings.Contains(t.PkgPath(), "k8s.io/kubernetes/")
```

Comment:
Any better idea? Will move all of these into a test package. IMO this is good enough for tests as they fall over if this test is wrong.